### PR TITLE
Mi-1278: Add Comments and lint fix

### PR DESCRIPTION
**Pull request by hrishavkmr**

### CODE REVIEW

File: app/lti.go [hunk 1]
```diff
@@ -4,15 +4,22 @@
...
 
 	"github.com/mattermost/mattermost-server/v5/model"
 	"github.com/mattermost/mattermost-server/v5/shared/mlog"
 )
 
-func (a* App) GetLTISettings() (*model.LTISettings, error) {
+//
+//	GetLTISettings() reads the LTI Config from Plugin Config
+//	and unmarshal the JSON to LTISettings struct.
+//
```
Comment (chetanyakan) [inline app/lti.go]: Explain what happens when
- LTI plugin is not installed
- LTI plugin is enabled/disabled

File: app/lti.go [hunk 2]
```diff
@@ -21,20 +28,26 @@ func (a* App) GetLTISettings() (*model.LTISettings, error) {
...
+		return nil, fmt.Errorf("Error marshaling LTI Config: %s", err.Error())
 	}
 
+	//
+	//	Unmarshal the Plugin Config to LTISettings
+	//
 	var LTISettings *model.LTISettings
 	if err = json.Unmarshal(configJson, &LTISettings); err != nil {
-		return nil, fmt.Errorf("Error unmarshaling LTI Config from json: %s",err.Error())
+		return nil, fmt.Errorf("Error unmarshaling LTI Config from json: %s", err.Error())
```
Comment (chetanyakan) [inline app/lti.go]: use `errors.Wrap` instead

File: config/client.go [hunk 3]
```diff
@@ -310,7 +310,13 @@ func GenerateLimitedClientConfig(c *model.Config, telemetryID string, license *m
 	props["GuestAccountsEnforceMultifactorAuthentication"] = strconv.FormatBool(*c.GuestAccountsSettings.EnforceMultifactorAuthentication)
 
 	props["EnableSignUpWithLTI"] = strconv.FormatBool(false)
+	//
+	//	Check that if LTI Plugin is installed.
+	//
 	if LTIConfig, ok := c.PluginSettings.Plugins[model.LTI_PLUGIN_ID]; ok {
+		//
+		//	Check that if LTI Authentication is enabled.
+		//
```
Comment (chetanyakan) [inline config/client.go]: You don't have to add comments about things that can be easily understood from readable code.

Instead, explain why these changes are needed.

E.g. here, you could mention what this code block does, i.e. add a client config parameter to allow the webapp to check if signup with LTI is enabled.

File: model/lti.go [hunk 4]
```diff
@@ -23,6 +23,7 @@ const (
 
 	LTI_USER_ID_PROP_KEY = "lti_user_id"
 
+	// plugin id of lti configuration plugin
```
Comment (chetanyakan) [inline model/lti.go]: 1. Comment over constant should start with constant name. 
2. mention what this plugin is used for.
e.g. LTI_PLUGIN_ID is the ID of the plugin used to maintain the LTI settings.

File: web/lti.go [hunk 5]
```diff
@@ -21,15 +21,17 @@ func (w *Web) InitLti() {
 
 func loginWithLTI(c *Context, w http.ResponseWriter, r *http.Request) {
 	mlog.Debug("Received an LTI Login request")
-
-	LTISettings, ltiErr := c.App.GetLTISettings();
-	if(ltiErr != nil){
-		mlog.Error(ltiErr.Error());
+	//
+	// Get the LTI Settings from plugin config.
+	//
```
Comment (chetanyakan) [inline web/lti.go]: this comment is not helpful. it's better if you remove this and similar comments.

**Review by chetanyakan — CHANGES_REQUESTED** [review]

**Review by manojmalik20 — APPROVED** [review]

File: app/lti.go (hunk 1 above)
Comment (chetanyakan) [reply app/lti.go]: what about this?

**Review by chetanyakan — APPROVED** [review]

**Pull request merged**